src/samplers/sampler.jl

```diff
@@ -32,7 +33,7 @@ assume(spl::Void, dist::Distribution, vn::VarName, vi::VarInfo) = begin
   if haskey(vi, vn)
     r = vi[vn]
   else
-    r = rand(dist)
+    r = init(dist)
```

Comment:
I see. Perhaps it's time to introduce a `Init` sampler type that takes care of initialisation. Then we can get rid of void samplers from our code, and 

```
type Init{T<:InferenceAlgorithm} end

# then for HMC
Init{HMC}

# for samplers using prior as a proposal
Init{SMC} # or Init{IS}, Init{PG}.
```